navigator_engine/api.py

```diff
@@ -32,18 +32,24 @@ def decide():
 
     graph = load_graph(choose_graph(input_data['data']['url']))
     data_loader = choose_data_loader(input_data['data']['url'])
-    data = data_loader('url', 'authorization_header', input_data['data'])
+    source_data = input_data['data']
     skip_actions = input_data.get('skipActions', [])
 
-    engine = DecisionEngine(graph, data, skip=skip_actions)
+    engine = DecisionEngine(
+        graph,
+        source_data,
+        data_loader=data_loader,
+        skip=skip_actions
+    )
     engine.decide()
     del engine.decision['node']
 
     return jsonify({
         "decision": engine.decision,
         "actions": engine.progress.action_breadcrumbs,
         "skippedActions": engine.progress.skipped,
-        "progress": engine.progress.report,
+        "removeSkips": engine.remove_skips,
```

Comment:
Maybe a better name `"removeSkippedActions"` to be consistent with the name `"skippedActions"`.